tests/acceptance/premium/test_branches_users_case_insensitive.py

```diff
@@ -0,0 +1,58 @@
+import pytest
+
+from gitlabform.gitlab import AccessLevel
+from tests.acceptance import (
+    run_gitlabform,
+    randomize_case,
+    get_only_branch_access_levels,
+)
+
+pytestmark = pytest.mark.requires_license
```

Comment:
What's the purpose of this test file (`tests/acceptance/premium/test_branches_users_case_insensitivity.py`)? Please correct me if I'm wrong here. I don't think username case-sensitivity is a gitlab premium/ultimate feature.

Reply:
Added in as an additional test for the `tests/acceptance/premium/test_branches.py` file, which also exists under premium features. Protected branches have _some_ premium protections on them - this does affect the permissions for `allowed_to_merge` and `allowed_to_push` as they both do username/group name lookups.

Reply:
Tried moving the branch case test to standard and removing the `requires_license` fixture, and it failed due to differences with the enterprise API

Reply:
> Protected branches have some premium protections on them - this does affect the permissions for allowed_to_merge and allowed_to_push as they both do username/group name lookups.

Ahh... that makes sense now. I didn't think of that. Thanks for clarifying it.